src/cmd_all/src/single_node.rs

```diff
@@ -65,41 +39,74 @@ pub struct SingleNodeOpts {
 
     /// The store directory used by meta store and object store.
     #[clap(long, env = "RW_SINGLE_NODE_STORE_DIRECTORY")]
-    pub store_directory: Option<String>,
-
-    /// The address of the meta node.
-    #[clap(long, env = "RW_SINGLE_NODE_META_ADDR")]
-    meta_addr: Option<String>,
-
-    /// The address of the compute node
-    #[clap(long, env = "RW_SINGLE_NODE_COMPUTE_ADDR")]
-    compute_addr: Option<String>,
-
-    /// The address of the frontend node
-    #[clap(long, env = "RW_SINGLE_NODE_FRONTEND_ADDR")]
-    frontend_addr: Option<String>,
+    store_directory: Option<String>,
 
-    /// The address of the compactor node
-    #[clap(long, env = "RW_SINGLE_NODE_COMPACTOR_ADDR")]
-    compactor_addr: Option<String>,
+    #[clap(flatten)]
+    node_opts: NodeSpecificOpts,
 }
 
-pub fn make_single_node_sql_endpoint(store_directory: &String) -> String {
-    format!(
-        "sqlite://{}/meta_store/single_node.db?mode=rwc",
-        store_directory
-    )
+/// # Node-Specific Options
+///
+/// ## Which node-specific options should be here?
+///
+/// 1. An option doesn't make sense for single-node deployment should not be here.
+///    - e.g. `meta_addr` will be automatically set to localhost.
+/// 2. An option that is hidden from help message and can be set from config file
+///    - e.g. `etcd_endpoints` is encouraged to be set from config file or environment variables.
+/// 3. An option that is only for cloud deployment should not be here.
+///    - e.g. `proxy_rpc_endpoint` is used in cloud deployment.
+///
+/// ## How to add an new option here?
+///
+/// Options for the other nodes are defined with following convention:
+///
+/// 1. The option name is either the same as the node name or with a prefix to indicate the node type.
+/// 2. The option doesn't have a default value and must be `Option<T>`, so that the default value in the node's `Opts` struct can be used.
+/// 3. The option doesn't need to read from environment variables, which will be done in the node's `Opts` struct.
+#[derive(Eq, PartialOrd, PartialEq, Debug, Clone, Parser)]
+pub struct NodeSpecificOpts {
```

Comment:
Should we add a prefix (e.g., `compute`) before the opts? It seems not very necessary though.

Reply:
The number of opts is indeed quite few. So the PR's idea LGTM

Reply:
> Should we add a prefix (e.g., `compute`) before the opts? It seems not very necessary though.

When I wrote that instructions, I thought there may be some, but actually not.